### PR TITLE
feat: enable configuration of the tls parameter for the mysql connection. i.e. tls=preferred

### PR DESCRIPTION
**Pull request by embroede**
## Description

Enable configuration of the tls parameter for the mysql connection. An example would be setting tls to "preferred", or "true". This is separate from the tlsConfig (renamed from tls) parameter in manager.yaml, which is used to setup a custom tls config, where tls key/certs are specified. This will fix the case where the manager cannot connect to a server that requires TLS.

See the tls parameter section in the below link:
https://pkg.go.dev/github.com/go-sql-driver/mysql#section-readme

## Related Issue

https://github.com/dragonflyoss/Dragonfly2/issues/1299

## Motivation and Context

Currently Manager cannot connect to a MySQL database that requires TLS, without fully configuring certs/keys (functionality added in https://github.com/dragonflyoss/Dragonfly2/pull/1015). By making this configurable, TLS can be used (even with only password authentication).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by gaius-qi** [comment]
Can tls be changed to manager configuration?

https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/config/config.go#L71

**Comment by embroede** [comment]
@gaius-qi  Yes I like that better, I will make the change.

**Comment by gaius-qi** [comment]
> @gaius-qi Yes I like that better, I will make the change.

Thx

**Comment by embroede** [comment]
@gaius-qi  I have added it to the configuration, and created a PR to update the docs here: https://github.com/dragonflyoss/d7y.io/pull/79

**Comment by gaius-qi** [comment]
You can refer to harbor, TLSConfig is used to set the SSL mode, do not rename the TLS configuration. 

When ssl mode is preferred and skip-verify set InsecureSkipVerify to true.

https://github.com/goharbor/harbor/blob/main/src/vendor/github.com/go-sql-driver/mysql/dsn.go#L48

File: manager/config/config.go [hunk 1]
```diff
@@ -87,8 +87,11 @@ type MysqlConfig struct {
 	// Enable migration
 	Migrate bool `yaml:"migrate" mapstructure:"migrate"`
 
-	// TLS configuration
-	TLS *TLSConfig `yaml:"tls" mapstructure:"tls"`
+    // TLS setting (can be one of "true", "false", "skip-verify",  or "preferred")
+    TLS string `yaml:"tls" mapstructure:"tls"`
```
Comment (gaius-qi) [inline manager/config/config.go:91]: You can refer to harbor, TLSConfig is used to set the SSL mode, do not rename the TLS configuration.

https://github.com/goharbor/harbor/blob/main/src/vendor/github.com/go-sql-driver/mysql/dsn.go#L48

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: manager/config/config_test.go [hunk 2]
```diff
@@ -51,7 +51,8 @@ func TestManagerConfig_Load(t *testing.T) {
 				Host:     "foo",
 				Port:     3306,
 				DBName:   "foo",
-				TLS: &TLSConfig{
+                TLS: "preferred",
```
Comment (gaius-qi) [inline manager/config/config_test.go:54]: Code needs format.

File: manager/database/mysql.go [hunk 3]
```diff
@@ -80,17 +80,19 @@ func formatDSN(cfg *config.MysqlConfig) (string, error) {
...
 		if err != nil {
 			return "", err
 		}
 
 		if err := mysql.RegisterTLSConfig("custom", tls); err != nil {
 			return "", err
 		}
-	}
+	} else if cfg.TLS != "" { // If no custom config is specified, use tls parameter if it is set
+        mysqlCfg.Params = map[string]string{"tls": cfg.TLS}
```
Comment (gaius-qi) [inline manager/database/mysql.go:94]: Code needs format.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: manager/config/config.go (hunk 1 above)
Comment (embroede) [reply manager/config/config.go]: Sure good point, fixed in latest commit.

File: manager/config/config_test.go (hunk 2 above)
Comment (embroede) [reply manager/config/config_test.go]: fixed

File: manager/database/mysql.go (hunk 3 above)
Comment (embroede) [reply manager/database/mysql.go]: fixed

Comment (gaius-qi) [reply manager/database/mysql.go]: You can refer to this https://github.com/go-sql-driver/mysql/issues/899。

Use mysql config's `TLSConfig`.

Comment (embroede) [reply manager/database/mysql.go]: Fixed :)

Comment (embroede) [reply manager/database/mysql.go]: @gaius-qi If everything is ok, could this be included in 2.0.4?

Comment (gaius-qi) [reply manager/database/mysql.go]: > @gaius-qi If everything is ok, could this be included in 2.0.4?

Fix the lint workflow and the code implementation should be fine.

Comment (embroede) [reply manager/database/mysql.go]: @gaius-qi  Thanks, it should be fixed now.

**Comment by gaius-qi** [comment]
Please fix unit tests. @embroede 

```text
2022-06-27T12:42:43.5657961Z === RUN   TestManagerConfig_Load
2022-06-27T12:42:43.5658567Z     config_test.go:107: 
2022-06-27T12:42:43.5659014Z         	Error Trace:	config_test.go:107
2022-06-27T12:42:43.5659405Z         	Error:      	Not equal: 
2022-06-27T12:42:43.5662766Z         	            	expected: &config.Config{Options:base.Options{Console:false, Verbose:false, PProfPort:0, Telemetry:base.TelemetryOption{Jaeger:"", ServiceName:""}}, Server:(*config.ServerConfig)(0xc00005b840), Database:(*config.DatabaseConfig)(0xc000338330), Cache:(*config.CacheConfig)(0xc000338340), ObjectStorage:(*config.ObjectStorageConfig)(0xc000012900), Metrics:(*config.MetricsConfig)(0xc00004f4c0)}
2022-06-27T12:42:43.5666560Z         	            	actual  : &config.Config{Options:base.Options{Console:false, Verbose:false, PProfPort:0, Telemetry:base.TelemetryOption{Jaeger:"", ServiceName:""}}, Server:(*config.ServerConfig)(0xc00005bb00), Database:(*config.DatabaseConfig)(0xc000339610), Cache:(*config.CacheConfig)(0xc000339b20), ObjectStorage:(*config.ObjectStorageConfig)(0xc000012fc0), Metrics:(*config.MetricsConfig)(0xc00004fc80)}
2022-06-27T12:42:43.5667595Z         	            	
2022-06-27T12:42:43.5667961Z         	            	Diff:
2022-06-27T12:42:43.5668504Z         	            	--- Expected
2022-06-27T12:42:43.5668899Z         	            	+++ Actual
2022-06-27T12:42:43.5669429Z         	            	@@ -33,3 +33,3 @@
2022-06-27T12:42:43.5669999Z         	            	    Migrate: (bool) false,
2022-06-27T12:42:43.5670858Z         	            	-   TLSConfig: (string) (len=9) "preferred",
2022-06-27T12:42:43.5671408Z         	            	+   TLSConfig: (string) "",
2022-06-27T12:42:43.5671986Z         	            	    TLS: (*config.TLSConfig)({
2022-06-27T12:42:43.5672378Z         	Test:       	TestManagerConfig_Load
2022-06-27T12:42:43.5672763Z --- FAIL: TestManagerConfig_Load (0.01s)
2022-06-27T12:42:43.5673014Z FAIL
```

**Comment by embroede** [comment]
@gaius-qi Sorry about that, fixed.

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by cndoit18 — APPROVED** [review]
LGTM

**Review by hyy0322 — APPROVED** [review]
LGTM

**Pull request merged**